pkg/applier/inventory/inventory_converter_test.go

```diff
@@ -0,0 +1,643 @@
+						GroupKind: schema.GroupKind{
+							Group: "obj2-group",
+							Kind:  "obj2-kind",
+						},
+					},
+				},
+				objStatuses: []actuation.ObjectStatus{
+					{
+						ObjectReference: actuation.ObjectReference{
+							Name:      "obj1-name",
+							Namespace: "obj1-ns",
+							Group:     "obj1-group",
+							Kind:      "obj1-kind",
+						},
+						Strategy:   actuation.ActuationStrategyApply,
+						Actuation:  actuation.ActuationSucceeded,
+						Reconcile:  actuation.ReconcileSucceeded,
+						UID:        "obj1-uid",
+						Generation: int64(21),
+					},
+					{
+						ObjectReference: actuation.ObjectReference{
+							Name:      "obj2-name",
+							Namespace: "obj2-ns",
+							Group:     "obj2-group",
+							Kind:      "obj2-kind",
+						},
+						Strategy:   actuation.ActuationStrategyApply,
+						Actuation:  actuation.ActuationPending,
+						Reconcile:  actuation.ReconcilePending,
+						UID:        "obj2-uid",
+						Generation: int64(12),
+					},
+				},
+			}.build(t),
+			wantObj: resourceGroupFactory{
+				spec: v1alpha1.ResourceGroupSpec{
+					Resources: []v1alpha1.ObjMetadata{
+						{
+							Name:      "obj1-name",
+							Namespace: "obj1-ns",
+							GroupKind: v1alpha1.GroupKind{
+								Group: "obj1-group",
+								Kind:  "obj1-kind",
+							},
+						},
+						{
+							Name:      "obj2-name",
+							Namespace: "obj2-ns",
+							GroupKind: v1alpha1.GroupKind{
+								Group: "obj2-group",
+								Kind:  "obj2-kind",
+							},
+						},
+					},
+				},
+				status: v1alpha1.ResourceGroupStatus{
+					ObservedGeneration: invGeneration,
+					Conditions: []v1alpha1.Condition{
+						{
+							Type:   v1alpha1.Stalled,
+							Status: v1alpha1.FalseConditionStatus,
+						},
+					},
+					ResourceStatuses: []v1alpha1.ResourceStatus{
+						{
+							ObjMetadata: v1alpha1.ObjMetadata{
+								Name:      "obj1-name",
+								Namespace: "obj1-ns",
+								GroupKind: v1alpha1.GroupKind{
+									Group: "obj1-group",
+									Kind:  "obj1-kind",
+								},
+							},
+							Status:    v1alpha1.Unknown,
+							Strategy:  v1alpha1.Apply,
+							Actuation: v1alpha1.ActuationSucceeded,
+							Reconcile: v1alpha1.ReconcileSucceeded,
+						},
+						{
+							ObjMetadata: v1alpha1.ObjMetadata{
+								Name:      "obj2-name",
+								Namespace: "obj2-ns",
+								GroupKind: v1alpha1.GroupKind{
+									Group: "obj2-group",
+									Kind:  "obj2-kind",
+								},
+							},
+							Status:    v1alpha1.Unknown,
+							Strategy:  v1alpha1.Apply,
+							Actuation: v1alpha1.ActuationPending,
+							Reconcile: v1alpha1.ReconcilePending,
+						},
+					},
+				},
+			}.build(t),
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			ic := NewInventoryConverter(declared.RootScope, invName, metadata.StatusEnabled)
+			rg, err := ic.InventoryToUnstructured(tc.fromObj, tc.toInv)
+			if tc.wantErr != nil {
+				require.Equal(t, tc.wantErr, err)
```

Comment:
I'm kinda surprised this works. This is why I wrote testerrors.AssertEqual, which doesn't require the error pointer to be exactly the same. But maybe that's only a problem on wrapped errors and not the top one.